### PR TITLE
Don't add new lines after unless present in graphql

### PR DESCRIPTION
**Pull request by ad1992**
Fixes https://github.com/prettier/prettier/issues/4164

### CODE REVIEW

File: tests/multiparser_js_graphql/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -386,6 +388,8 @@ gql\`
     test
   }
 
+
+
```
Comment (lydell) [inline tests/multiparser_js_graphql/__snapshots__/jsfmt.spec.js.snap]: This looks like a regression?

File: src/language-graphql/printer-graphql.js [hunk 2]
```diff
@@ -24,10 +24,24 @@ function genericPrint(path, options, print) {
 
   switch (n.kind) {
     case "Document": {
-      return concat([
-        join(concat([hardline, hardline]), path.map(print, "definitions")),
-        hardline
-      ]);
+      const parts = [];
+      let i = 0;
+      path.map(pathChild => {
```
Comment (lydell) [inline src/language-graphql/printer-graphql.js]: The index is provided as the second parameter to `.map` so no need to manually count.

File: tests/multiparser_js_graphql/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -17,6 +17,8 @@ gql\`
 
 `;
 
+
+
```
Comment (lydell) [inline tests/multiparser_js_graphql/__snapshots__/jsfmt.spec.js.snap]: Just a curios Jest question: Does anybody know where these newlines came from?

Comment (ad1992) [reply tests/multiparser_js_graphql/__snapshots__/jsfmt.spec.js.snap]: This got fixed when i merged with latest master.

**Comment by ad1992** [comment]
@lydell @j-f1  can this be merged ?

**Comment by j-f1** [comment]
Thanks for contributing!

**Pull request merged**